src/annotator/anchoring/test/range-test.js

```diff
@@ -0,0 +1,532 @@
+        assert.isTrue(result instanceof Range.SerializedRange);
+        assert.deepEqual(result, {
+          start: '/section[1]/p[1]',
+          startOffset: 0,
+          end: '/section[1]/span[1]/p[1]',
+          endOffset: 1,
+        });
+      });
+
+      it('converts BrowserRange to SerializedRange instance with `ignoreSelector` condition', () => {
+        const browserRange = createBrowserRange();
+        const result = browserRange.serialize(container, '#p-3');
+        // End xpath shall ignore the provided selector.
+        assert.deepEqual(result, {
+          start: '/section[1]/p[1]',
+          startOffset: 0,
+          end: '/section[1]/span[1]',
+          endOffset: 1,
+        });
+      });
+    });
+  });
+
+  describe('NormalizedRange', () => {
+    describe('#limit, #text and #textNodes methods', () => {
+      it('does not limit range if the limit node resides outside of bounds', () => {
+        const limit = createNormalizedRange().limit(
+          container.querySelector('#span-2')
+        );
+        assert.isNull(limit);
+      });
+
+      [
+        {
+          bound: '#p-1',
+          textNodes: ['text 1'],
+        },
+        {
+          bound: '#p-2',
+          textNodes: ['text 2a', 'text 2b'],
+        },
+        {
+          bound: '#section-1',
+          textNodes: ['text 1', 'text 2a', 'text 2b', 'text 3'],
+        },
+      ].forEach(test => {
+        it('limits range to the bounding node', () => {
+          let limit = createNormalizedRange().limit(
+            container.querySelector(test.bound)
+          );
+          assert.equal(limit.text(), test.textNodes.join(''));
+          // To get a node value from jquery, use ".data".
+          assert.deepEqual(
+            test.textNodes,
+            limit.textNodes().map(n => n.data)
+          );
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      let fakeSetStartBefore;
+      let fakeSetEndAfter;
+
+      beforeEach(() => {
+        sinon.stub(document, 'createRange');
+        fakeSetStartBefore = sinon.stub();
+        fakeSetEndAfter = sinon.stub();
+        document.createRange.returns({
+          setStartBefore: fakeSetStartBefore,
+          setEndAfter: fakeSetEndAfter,
+        });
+      });
+
+      afterEach(() => {
+        document.createRange.restore();
+      });
+
+      it('converts normalized range to native range', () => {
+        const range = createNormalizedRange();
+        const nativeRange = range.toRange();
+        assert.deepEqual(nativeRange, document.createRange());
+        assert.calledWith(fakeSetStartBefore, range.start);
+        assert.calledWith(fakeSetEndAfter, range.end);
+      });
+    });
+
+    describe('#normalize', () => {
+      it('returns an instance of itself', () => {
+        const initialRange = createNormalizedRange();
+        const normalizedRange = initialRange.normalize();
+        assert.deepEqual(initialRange, normalizedRange);
```

Comment:
```suggestion
        assert.equal(initialRange, normalizedRange);
```

More than just "an instance of `NormalizedRange`" it returns just `this`.